src/Microsoft.OData.Core/JsonLight/ODataJsonLightBatchReader.cs

```diff
@@ -53,6 +53,18 @@ internal sealed class ODataJsonLightBatchReader : ODataBatchReader
         /// </summary>
         private ODataJsonLightBatchPayloadItemPropertiesCache messagePropertiesCache = null;
 
+        /// <summary>
+        /// Dictionary for keeping track of each request's associated atomic group id, which is null
+        /// for request that does not belong to atomic group.
+        /// </summary>
+        private Dictionary<string, string> requestIdToAtomicGroupId = new Dictionary<string, string>();
+
+        /// <summary>
+        /// Dictionary for keeping track of each atomic group's member request id. This is optimization
+        /// for reversed lookup.
+        /// </summary>
+        private Dictionary<string, IList<string>> atomicityGroupIdToRequestId = new Dictionary<string, IList<string>>();
+
         /// <summary>

```

Comment:
Why do you need dictionaries, it looks like you just check to see if the dictionary contains the key, not the relationship between the id and the atomicity group. Can you just use a list?

Reply:
I have changed it to a HashSet.